optuna/samplers/_nsga2/crossover.py

```diff
@@ -0,0 +1,456 @@
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Sequence
+
+import numpy as np
+
+from optuna._transform import _SearchSpaceTransform
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalDistribution
+from optuna.distributions import DiscreteUniformDistribution
+from optuna.distributions import IntLogUniformDistribution
+from optuna.distributions import IntUniformDistribution
+from optuna.distributions import LogUniformDistribution
+from optuna.distributions import UniformDistribution
+from optuna.study import Study
+from optuna.study import StudyDirection
+from optuna.trial import FrozenTrial
+
+
+_NUMERICAL_DISTRIBUTIONS = (
+    UniformDistribution,
+    LogUniformDistribution,
+    DiscreteUniformDistribution,
+    IntUniformDistribution,
+    IntLogUniformDistribution,
+)
+
+
+def crossover(
+    crossover_name: str,
+    study: Study,
+    parent_population: Sequence[FrozenTrial],
+    search_space: Dict[str, BaseDistribution],
+    rng: np.random.RandomState,
+    swapping_prob: float,
+    dominates: Callable[[FrozenTrial, FrozenTrial, Sequence[StudyDirection]], bool],
+) -> Dict[str, Any]:
+
+    while True:  # Repeat while parameters lie outside search space boundaries.
+        parents = _selection(crossover_name, study, parent_population, rng, dominates)
+        child = {}
+        transes = []
+        distributions = []
+        param_names = []
+        parents_not_categorical_params: List[List[np.float64]] = [[] for _ in range(len(parents))]
+        for param_name in search_space.keys():
+            param_distribution = search_space[param_name]
+            parents_param = [p.params[param_name] for p in parents]
```

Comment:
It seems that the plural form is appropriate.
```suggestion
            parent_params = [p.params[param_name] for p in parents]
```